src/Microsoft.OData.Core/Json/DefaultStreamBasedJsonWriterFactory.cs

```diff
@@ -0,0 +1,35 @@
+﻿//---------------------------------------------------------------------
+// <copyright file="DefaultStreamBasedJsonWriterFactory.cs" company="Microsoft">
+//      Copyright (C) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.
+// </copyright>
+//---------------------------------------------------------------------
+
+#if NETCOREAPP3_1_OR_GREATER
+using System;
+using System.IO;
+using System.Text;
+
+namespace Microsoft.OData.Json
+{
+
+    /// <summary>
+    /// Default factory for JSON writer that writes directly to a stream
+    /// rather than a TextWriter.
+    /// </summary>
+    [CLSCompliant(false)]
+    public sealed class DefaultStreamBasedJsonWriterFactory : IStreamBasedJsonWriterFactory
+    {
+        public IJsonWriter CreateJsonWriter(Stream stream, bool isIeee754Compatible, Encoding encoding)
+        {
+            // TODO: only supports Utf8-encoding for now. Support for other encodings

```

Comment:
Can you link to an issue instead of a TODO in the code?

Reply:
Sure. I've created the issue here: https://github.com/OData/odata.net/issues/2419